src/syntax/ast.rs

```diff
@@ -1533,10 +1533,7 @@ impl Closure {
     ///
     /// This only exists if you use the function syntax sugar: `let f(x) = y`.
     pub fn name(&self) -> Option<Ident> {
-        match self.0.cast_first_match::<Pattern>()?.kind() {
-            PatternKind::Ident(ident) => Some(ident),
-            _ => Option::None,
-        }
+        self.0.children().next()?.cast()
```

Comment:
is this intentionally not `self.0.cast_first_match()` because it should only apply if it's the child at index 0?

Reply:
Yes, that fails unnamed closures `(x, y) => f`. This is the way it was before #532.